### PR TITLE
Editor zooming

### PR DESCRIPTION
**Pull request by Vankata453**
By holding CTRL and scrolling with the mouse, the camera in the editor can now be zoomed in and out. There is a set zoom limit: `0.5` - `3.0`.

When zooming in, the camera position will progressively focus on the position of the mouse, relative to the screen center.

Additionally, "src/editor/util.*" has been removed, since it only provided an unneeded `draw_tile` wrapper function.

Mentioned in #2421.

### CODE REVIEW

**Comment by mrkubax10** [comment]
It doesn't seem right :/

https://github.com/SuperTux/supertux/assets/63350514/da4a428a-b0db-4ebb-8e88-34dff68040d9

**Comment by Vankata453** [comment]
> It doesn't seem right :/

You mean where it goes outside the sector border?

**Comment by mrkubax10** [comment]
Yeah and also weird snapping while zooming but it's not visible on the video

**Comment by Vankata453** [comment]
> Yeah and also weird snapping while zooming but it's not visible on the video

Not sure how to do smooth zooming. Maybe a solution would be to decrease the zoom sensitivity further.

**Comment by Rusty-Box** [comment]
Can we have a way to tell how far in/out you zoomed in? Like a "x1.0", "x2.0" etc somewhere
I'll give this a closer look soon

**Comment by Rusty-Box** [comment]
> Can we have a way to tell how far in/out you zoomed in? Like a "x1.0", "x2.0" etc somewhere I'll give this a closer look soon

This should still be added, also, would be nice to have a keyboard shortcut to instantly go back to a x1.0 zoom.

**Review by Rusty-Box — APPROVED** [review]
Looks good!

File: src/editor/overlay_widget.cpp [hunk 1]
```diff
@@ -1487,7 +1491,20 @@ EditorOverlayWidget::draw(DrawingContext& context)
...
+  if (scale_percentage != 100)
+    context.color().draw_text(Resources::big_font, std::to_string(scale_percentage) + '%',
+                              Vector(context.get_width() - 140.f, 15.f),
+                              ALIGN_RIGHT, LAYER_OBJECTS + 1, Color::WHITE);
+
+  context.push_transform();
+  context.set_translation(m_editor.get_sector()->get_camera().get_translation());
+  context.transform().scale = scale;
+
+  draw_path(context);
```
Comment (MatusGuy) [inline src/editor/overlay_widget.cpp:1507]: what

**Review by MatusGuy — APPROVED** [review]

File: src/editor/overlay_widget.cpp (hunk 1 above)
Comment (Vankata453) [reply src/editor/overlay_widget.cpp:1507]: Please be more verbose with comments.

**Review by tobbi — APPROVED** [review]
I have no idea what I'm doing with my life.

**Comment by Frostwithasideofsalt** [comment]
time 2 test

**Review by Frostwithasideofsalt — APPROVED** [review]

**Pull request merged**